sw-sysemu/memory/pma.h

```diff
@@ -0,0 +1,33 @@
+/*-------------------------------------------------------------------------
+* Copyright (c) 2025 Ainekko, Co.
+* SPDX-License-Identifier: Apache-2.0
+*-------------------------------------------------------------------------*/
+
+#ifndef BEMU_PMA_H
+#define BEMU_PMA_H
+
+#include "emu_defines.h"
+#include "processor.h"
+
+namespace bemu {
+
+uint64_t pma_check_data_access(const Hart& cpu, uint64_t vaddr,
+                                      uint64_t addr, size_t size,
+                                      mem_access_type macc,
+                                      mreg_t mask = mreg_t(-1),
+                                      cacheop_type cop = CacheOp_None);
+
+uint64_t pma_check_fetch_access(const Hart& cpu, uint64_t vaddr,
+                                       uint64_t addr, size_t size);
+
+uint64_t vmemtranslate(const Hart& cpu, uint64_t vaddr, size_t size,
+                              mem_access_type macc);
+
+#if EMU_ETSOC1
```

Comment:
I mean, we have the same kind of expectation of page table walk in erbium, we don't have to remove it, even if we don't use it.